File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math
from itertools import permutations
def solution(numbers):
    answer = set()
    numbers = list(numbers)
    
    for cnt in range(1, len(numbers)+1):
        
        for case in permutations(numbers, cnt):
            case_num = int(''.join(case))
        
            if case_num not in answer and is_sosu(case_num): 
                answer.add(case_num)
    
    return len(answer)


def is_sosu(num):
    
    if num == 0 or num ==1:
        return False
    if num == 2:
        return True
    
    for i in range(2,int(math.sqrt(num)) + 1):
        
        if num % i == 0:
            return False
        
    return True

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
import java.math.*;
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int n) {
        // int answer = 0;
        boolean[] answer = new boolean[n+1];
        double limit = Math.sqrt(n);
        answer[0] = true;
        answer[1] = true;

        for (int i=2; i< limit;i++) {
            for (int j=i+i;j<=n;j+=i) {
                answer[j] = true;
            }
        }
        int cnt = 0;
        for (int i=0; i < answer.length; i++) {
            if (answer[i] == false) {
                cnt += 1;
            }
        }
        return cnt;
    }
}



